cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -83,7 +96,10 @@ def dist_on_device(self, lq1: 'cirq.Qid', lq2: 'cirq.Qid') -> int:
         Returns:
             The shortest path distance.
         """
-        return len(self._physical_shortest_path(self._map[lq1], self._map[lq2])) - 1
+        source = self._qubit_to_int[self._map[lq1]]
+        target = self._qubit_to_int[self._map[lq2]]
```

Comment:
@tanujkhattar I'm still doing 4 dictionary accesses here to go from logical to physical to integer. Can maybe instead store a dict from logical to int directly but not sure if that will help.